### PR TITLE
Added MangaSee presence

### LINKED ISSUE #3445: Mangasee | mangasee123.com
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 

Wisty#2774

**What's the name of the service?** 

https://www.mangasee123.com/ | MangaSee

**What should the Presence display?** 

What manga you're currently reading and what chapter you're reading

**If possible, please provide a logo for the service (512x512 minimum)** 
mangasee

Or maybe

mangaseeses

### PR DESCRIPTION
**Pull request by MareGraphics**
Fully functional MangaSee123.com presence.
resolve #3445
Screenshots:
image
image
image
image

Settings off screenshots:
Buttons and icons:
image
Entries numbers: (on and off):
image
image

### CODE REVIEW

File: websites/M/MangaSee/presence.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        data.details = "Browsing all manga";
+    else if(pathname == "/discussion/")
+        data.details = "Viewing discussion page";
+    else if(pathname.endsWith("post.php")){
+      const title = document.querySelector(".BoxBody > h1").textContent,
+            author = document.querySelector(".Description > span").textContent,
+            link = window.location.href;
+      data.details = "Discussion: " + title;
+      data.state = "by " + author;
+      data.buttons = [{label: "Join discussion", url: link}];
```
Comment (ririxidev) [inline websites/M/MangaSee/presence.ts:36]:
```suggestion
      data.buttons = [{label: "View discussion", url: link}];
```

File: websites/M/MangaSee/presence.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    largeImageKey: "logo",
+    startTimestamp: browsingStamp
+  },
+    pathname = document.location.pathname;
+    if (pathname === "/"){
+      data.details = "Viewing the Homepage";
+    }
+    else if(pathname == "/search/" && window.location.search.substr(0,1) == "?"){
+      const query = document.location.search,
+      urlParams = new URLSearchParams(query),
```
Comment (ririxidev) [inline websites/M/MangaSee/presence.ts:20]:
```suggestion
      const urlParams = new URLSearchParams(document.location.search),
```

File: websites/M/MangaSee/presence.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        const link = window.location.href;
+        data.buttons = [{label: "View manga", url: link}];
+      }
+    }
+    else if(pathname.startsWith("/read-online/")){
+      const title = document.querySelector(".col-lg-4 > a").textContent.replace(new RegExp("\\\t","g"),'').replace(new RegExp("\\\n","g"),''),
+            chapter = document.querySelector('button[data-target="#ChapterModal"]').textContent.replace(new RegExp("\\\t","g"),'').replace(new RegExp("\\\n","g"),''),
+            page = document.querySelector('button[data-target="#PageModal"]').textContent.replace(new RegExp("\\\t","g"),'').replace(new RegExp("\\\n","g"),'');
+      data.details = title;
+      data.state = "📖Ch. " + chapter.split(" ")[1] + " 📄 " + page.split(" ")[1];
```
Comment (ririxidev) [inline websites/M/MangaSee/presence.ts:70]:
```suggestion
      data.state = "📖 Ch. " + chapter.split(" ")[1] + " 📄 " + page.split(" ")[1];
```

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/M/MangaSee/presence.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      const title = document.querySelector(".col-lg-4 > a").textContent.replace(new RegExp("\\\t","g"),'').replace(new RegExp("\\\n","g"),''),
+            chapter = document.querySelector('button[data-target="#ChapterModal"]').textContent.replace(new RegExp("\\\t","g"),'').replace(new RegExp("\\\n","g"),''),
+            page = document.querySelector('button[data-target="#PageModal"]').textContent.replace(new RegExp("\\\t","g"),'').replace(new RegExp("\\\n","g"),'');
+      data.details = title;
+      data.state = "📖 Ch. " + chapter.split(" ")[1] + " 📄 " + page.split(" ")[1];
+      data.smallImageKey = v2icons ? "read" : "read-v2";
+      if(buttons){
+        const link = window.location.href;
+        data.buttons = [{label: "View manga", url: link}];
+      }
```
Comment (Bas950) [inline websites/M/MangaSee/presence.ts:74]:
```suggestion
      if (buttons)  data.buttons = [{label: "View manga", url: window.location.href}];
```

File: websites/M/MangaSee/presence.ts [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    else if(pathname.startsWith("/manga/")){
+      const title = document.querySelector(".list-group-item > h1").textContent;
+      data.details = "Viewing manga:";
+      data.state = title;
+      data.smallImageKey = v2icons ? "view" : "view-v2";
+      if(buttons){
+        const link = window.location.href;
+        data.buttons = [{label: "View manga", url: link}];
+      }
```
Comment (Bas950) [inline websites/M/MangaSee/presence.ts:62]:
```suggestion
      if (buttons) data.buttons = [{label: "View manga", url: window.location.href}];
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — DISMISSED** [review]

File: websites/M/MangaSee/dist/metadata.json [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        "title": "Show buttons",
+        "icon": "fas fa-compress-arrows-alt",
+        "value": true
+      },
+      {
+        "id": "v2icons",
+        "title": "Circle icons",
+        "icon": "fas fa-circle",
+        "value": true
+      }
```
Comment (ririxidev) [inline websites/M/MangaSee/dist/metadata.json:39]: remove there these v2icons there too like in webtoon :/

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/M/MangaSee/dist/metadata.json (hunk 6 above)
Comment (MareGraphics) [reply websites/M/MangaSee/dist/metadata.json:39]: I will when I get back home :D i just wanted to allow some customization xD

Comment (ririxidev) [reply websites/M/MangaSee/dist/metadata.json:39]: Waiting.

Comment (MareGraphics) [reply websites/M/MangaSee/dist/metadata.json:39]: I'll do it tomorrow, was busy these days :)

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**